### PR TITLE
454 chan sel

### PR DESCRIPTION
**Pull request by dfsp-spirit**
Changes Summary
----------------
See #454. This is an investigation branch that currently mostly adds logging to identify the source of the slowdown. It identified best_match as the main source of the excessive runtime, which takes 0.5 sec per trial, at 180k trials, for this dataset, leading to ~ 25h runtime.

EDIT: `best_match` in the end wasn't the culprit, an ineffecient handling of the `.time` property was. For a lean channel selection, the costly `best_match` does not need to be invoked at all..

**We were able to avoid and speed up various internal function calls, and now the runtime is down from 25h to <10 minutes.**

Reviewer Checklist
------------------
- [x] Are testing routines present?
- [x] Do objects in the global package namespace perform proper parsing of their input? 
- [x] Are all docstrings complete and accurate?
- [ ] Is the CHANGELOG.md up to date?

### CODE REVIEW

**Comment by KatharineShapcott** [comment]
Apologies for the unsolicited advice, but couldn't you just temporarily remove the trial definition for a channel only selection and add it back in afterwards?

**Comment by tensionhead** [comment]
Yes we also thought along those lines! Yet I think it is still interesting to find out, why a mere channel selection requires that much time/computations..

**Comment by KatharineShapcott** [comment]
I suspect it might also be due to the way syncopy HDF5 files are stored, without chunking in the channel dimension I think you actually have to load in the whole file to access just one channel.

**Comment by dfsp-spirit** [comment]
The selection, as everything in Syncopy, happens via trials, so we have to enumerate all trials. This branch shows that the major part of the time required to do that goes into the `best_match()` function. I will, for now, leave it at that and close this. As Katherine suggested, the workaround is to select channels before applying a trial definition like that.

**Review by tensionhead — APPROVED** [review]
Great, together with #489 we made another performance leap quite noticable for people with >1 000 trials, and critical for people with >10 000 trials.

Compared to current `master`, selecting 3 channels and 1000 trials from a 40 channel 10 000 trial dataset now takes 0.5 seconds instead of 55 seconds.

File: syncopy/datatype/selector.py [hunk 1]
```diff
@@ -894,6 +883,7 @@ def _make_consistent(self, data):
 
                 # Update selector properties
                 for selection in actualSelections:
+                    print(getattr(self, "_{}".format(selection)), self._dataClass)
```
Comment (dfsp-spirit) [inline syncopy/datatype/selector.py:886]: there are commits missing, we have already removed this

**Pull request merged**